src/sdk/base/EtherspotWalletAPI.ts

```diff
@@ -69,8 +91,178 @@ export class EtherspotWalletAPI extends BaseAccountAPI {
     return accountContract.interface.encodeFunctionData('uninstallModule', [moduleTypeId, module, deinitData]);
   }
 
+  // Function to get all executors
+  async getAllExecutors(): Promise<string[]> {
+    // Assuming getExecutorsPaginated is a contract function that can be called
+    // Initialize variables
+    let pageSize = 50;
+    let lastAddress = SENTINEL_ADDRESS; // Assuming this is your SENTINEL value
+    let totalExecutors: string[] = [];
+
+    // Pagination loop
+    let tempExecutors: string[];
+    const accountContract = EtherspotWallet7579__factory.connect(await this.getAccountAddress(), this.provider);
+
+    do {
+      // Fetch a page of executors
+      [tempExecutors, lastAddress] = await accountContract.getExecutorsPaginated(lastAddress, pageSize);
+
+      // Append executors to the total list
+      totalExecutors = [...totalExecutors, ...tempExecutors];
+
+      // Break if it's the last page
+      if (tempExecutors.length < pageSize || lastAddress === SENTINEL_ADDRESS || lastAddress === ethers.constants.AddressZero) {
+        break;
+      }
+    } while (true);
+
+    return totalExecutors;
+  }
+
+  // Adjusting the function to be generic for handling different module types
+  async getPreviousAddressInSentinelList(targetAddress: string, moduleTypeId: MODULE_TYPE): Promise<string> {
+    let pageSize = 50;
```

Comment:
Should we pass this as a configurable parameter (with default value as 50)?

Reply:
updated with usage of DEFAULT_PAGE_SIZE and caller can override with the value they wish to paginate with